tests/apps/tests.py

```diff
@@ -528,3 +531,112 @@ def test_multiple_paths_explicit_path(self):
             with self.settings(INSTALLED_APPS=["nsapp.apps.NSAppConfig"]):
                 app_config = apps.get_app_config("nsapp")
                 self.assertEqual(app_config.path, self.app_path)
+
+
+class QueryPerformingAppTests(SimpleTestCase):
+    databases = {"default", "other"}
+
+    expected_message = (
+        "Accessing the database during app initialization is discouraged."
+    )
+
+    def tearDown(self) -> None:
+        TotallyNormal.objects.using("default").delete()
+        TotallyNormal.objects.using("other").delete()
+
+    def test_setup_does_not_initialize_database_connections(self):
+        self.reset_initialized_database_connections()
+
+        django.setup()
+
+        self.assertEqual(connections.all(initialized_only=True), [])
```

Comment:
Now, we can remove all things related with checking that connections are initialized as we don't do anything tricky in `setup()`.